### PR TITLE
Use new style checkboxes for plugin manager

### PR DESCRIPTION
**Pull request by janfaracik**
Small PR to use the new style checkboxes for the plugin manager.

**Before**
image

**After**
image

### Proposed changelog entries

* Use new style checkboxes for plugin manager.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Looks good, needs running through ATH though

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/PluginManager/available.jelly:60]: This used to tell users what the checkbox means. Now there's nothing left. Why?

File: core/src/main/resources/hudson/PluginManager/table.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/PluginManager/table.jelly:99]: Previously, `data` allowed users to restore the default sort order (by popularity) after changing it. This looks like it is no longer possible now.

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/PluginManager/available.jelly:60]: This used to tell users what the checkbox means. Now there's nothing left. Why?

File: core/src/main/resources/hudson/PluginManager/table.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/PluginManager/table.jelly:99]: From what I can see `table.jelly` is only ever used for the updates page so `isUpdates` is always `true`, so there's never actually any meaningful sorting.

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/PluginManager/available.jelly:60]: I think considering the average Jenkins user tends to be more of a power user the label isn't particularly beneficial.

My eventual plan is to move the select all button to its position, ala Gmail - 
image
And have the Select all/compatible/none be a dropdown from there.

File: core/src/main/resources/hudson/PluginManager/table.jelly
Comment (daniel-beck) [reply core/src/main/resources/hudson/PluginManager/table.jelly:99]: Interesting, it used to be for "updates" and "available", but that seems no longer the case.

https://github.com/jenkinsci/jenkins/blob/dacf078913d52a2e8c8a03f63d8401738ca317a7/core/src/main/resources/hudson/PluginManager/available.jelly#L64-L66 is probably a leftover from that heritage.

**Review by NotMyFault — COMMENTED** [review]
The checkbox column at the "Updates" view is much larger than on the "Available" view. Is that an intended change? 
Bildschirmfoto 2022-05-03 um 13 29 28
Bildschirmfoto 2022-05-03 um 13 29 48

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (daniel-beck) [reply core/src/main/resources/hudson/PluginManager/available.jelly:60]: I expect this will potentially confuse new admins (IIRC there's not even tooltips). What's the advantage of having no label?

**Comment by janfaracik** [comment]
> The checkbox column at the "Updates" view is much larger than on the "Available" view. Is that an intended change?
> Bildschirmfoto 2022-05-03 um 13 29 28
> Bildschirmfoto 2022-05-03 um 13 29 48

Not intentional! Fixed now :)

File: core/src/main/resources/hudson/PluginManager/available.jelly
Comment (janfaracik) [reply core/src/main/resources/hudson/PluginManager/available.jelly:60]: I personally think the 'Install' column header is confusing, if it were to have a header ideally it would be 'Select' as that's what the column actually does, the install button is completely separate to the table. Plus I think it looks cleaner without it.

Anyhoo, I've restored it 👍

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by NotMyFault** [comment]
Just noticed the following while testing other PRs:
https://user-images.githubusercontent.com/13383509/167621804-369eeeb3-69bc-482e-90df-41f59c6d1813.mp4

The checkboxes don't drop their active state, if you select them, as opposed to other locations.

**Comment by janfaracik** [comment]
> Just noticed the following while testing other PRs: https://user-images.githubusercontent.com/13383509/167621804-369eeeb3-69bc-482e-90df-41f59c6d1813.mp4
> 
> The checkboxes don't drop their active state, if you select them, as opposed to other locations.

Ah doesn't seem to do that on Safari - good spot. I've narrowed it down to it being the `for=""` attribute on the label, it doesn't do it for checkboxes with the 'attach-previous' class. I'll look into a fix for it 👍

**Pull request merged**
